lib/stdio/duplex.js

```diff
@@ -0,0 +1,42 @@
+import {Duplex, Readable, PassThrough, getDefaultHighWaterMark} from 'node:stream';
+
+/*
+Transform an array of generator functions into a `Duplex`.
+The `Duplex` is created by `Duplex.from()` made of a writable stream and a readable stream, piped to each other.
+- The writable stream is a simple `PassThrough`, so it only forwards data to the readable part.
+- The `PassThrough` is read as an iterable using `passThrough.iterator()`.
+- This iterable is transformed to another iterable, by applying the encoding generators.
+	Those convert the chunk type from `Buffer` to `string | Uint8Array` depending on the encoding option.
+- This new iterable is transformed again to another one, this time by applying the user-supplied generator.
+- Finally, `Readable.from()` is used to convert this final iterable to a `Readable` stream.
+ */
+export const generatorsToDuplex = (generators, {objectMode}) => {
+	const highWaterMark = getDefaultHighWaterMark(objectMode);
+	const passThrough = new PassThrough({objectMode, highWaterMark, destroy: destroyPassThrough});
+	let iterable = passThrough.iterator();
+
+	for (const generator of generators) {
+		iterable = generator(iterable);
+	}
+
+	const readableStream = Readable.from(iterable, {objectMode, highWaterMark});
+	const duplexStream = Duplex.from({writable: passThrough, readable: readableStream});
+	return duplexStream;
+};
```

Comment:
IMHO that's actually a fairly useful utility on its own, that might deserve its own package? It also does not have any external dependencies and is pretty much self-contained.

It relies on Node.js streams, although web streams can be produced too using [`Duplex.toWeb()`](https://nodejs.org/api/stream.html#streamduplextowebstreamduplex).

That could be useful for anyone wanting to transform streams but is too scared of using them. Generators are pretty straightforward.

Reply:
> IMHO that's actually a fairly useful utility on its own, that might deserve its own package? 🤔

Yeah, potentially. I think it would be good to let the code mature a bit here first though.

Also not clear to me how this utility is different from `stream.compose`. Might be worth documenting that.